src/components/NewDatePicker/index.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import {View, Animated} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CalendarPicker from '../CalendarPicker';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import * as Expensicons from '../Icon/Expensicons';
+import {propTypes as datePickerPropTypes, defaultProps as defaultDatePickerProps} from './datePickerPropTypes';
+
+const propTypes = {
+    ...datePickerPropTypes,
+};
+
+const datePickerDefaultProps = {
+    ...defaultDatePickerProps,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isPickerVisible: false,
+            selectedDate: moment(props.value || props.defaultValue).toDate(),
+        };
+
+        this.setDate = this.setDate.bind(this);
+        this.showPicker = this.showPicker.bind(this);
+        this.hidePicker = this.hidePicker.bind(this);
+
+        this.opacity = new Animated.Value(0);
+
+        /* We're using uncontrolled input otherwise it wont be possible to
+        * raise change events with a date value - each change will produce a date
+        * and make us reset the text input */
```

Comment:
```suggestion
        // We're using uncontrolled input otherwise it wont be possible to
        // raise change events with a date value - each change will produce a date
        // and make us reset the text input
```